src/main/java/com/google/starfish/servlets/UploadGNoteServlet.java

```diff
@@ -0,0 +1,122 @@
+package com.google.starfish.servlets;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;  
+import javax.servlet.http.Cookie;  
+import com.google.starfish.services.LabelService;
+import com.google.starfish.services.MiscNoteLabelService;
+import javax.sql.DataSource;
+import java.sql.Connection;  
+import java.sql.SQLException;  
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Date;
+import java.util.Calendar;
+
+/**
+ * Servlet to handle note uploading and adding associated labels to the database
+ * for gnotes from the chrome extension
+ */
+@WebServlet("/upload-gnote")
+public class UploadGNoteServlet extends HttpServlet {
+
+  private final String COOKIE_NAME = "SFCookie";
+  private LabelService labelService = new LabelService();
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    if(!validateUser(request)) {
+      response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      return;
+    }
+
+    HttpSession activeSession = request.getSession(false);
+    String userId = (String) activeSession.getAttribute("user_id");
+
+    String title = request.getParameter("title");
+    String school = request.getParameter("school").toLowerCase();
+    String course = request.getParameter("course").toLowerCase();
+    String sourceUrl = request.getParameter("sourceUrl");
+    String pdfSource = request.getParameter("pdfSource");
+
+    DataSource pool = (DataSource) request.getServletContext().getAttribute("my-pool");
+
+    // Insert all labels before inserting note due to foreign key constraint
+    addLabels(pool, school, course);
+    
+    try (Connection conn = pool.getConnection()) {
+      String stmt =
+          "INSERT INTO notes ( "
+              + "author_id,"
+              + "school,"
+              + "course,"
+              + "title,"
+              + "source_url,"
+              + "pdf_source,"
+              + "date_created,"
+              + "num_downloads ) "
+        + "VALUES ( "
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "?,"
+              + "0 ); ";
+
+      try (PreparedStatement noteStmt = conn.prepareStatement(stmt, new String[] {"id"})) {
```

Comment:
nit: You don't need the second param in `prepareStatement`, that was only added in the other servlet because I needed to get the `id` of the inserted row back by calling `generateKeys`.

Reply:
Ohh I see, I'll remove that!